include/aspect/particle/particle_handler.h

```diff
@@ -237,6 +422,50 @@ namespace aspect
         update_next_free_particle_index();
 
         /**
+         * Transfer particles that have crossed subdomain boundaries to other
+         * processors. The transfer occurs in two steps. As a first step all
+         * processes notify their neighbor processes how many particles will
+         * be sent to them. Because neighbor processes are defined as the owner
+         * of ghost cells of the current process, this also handles
+         * periodic boundaries correctly. Afterwards the transfer is done in the
+         * same way as local communication between neighbor processes.
+         * All received particles and their new cells will be appended to the
+         * @p received_particles vector.
+         *
+         * @param [in] sent_particles All particles that should be sent and
+         * their new subdomain_ids are in this map.
```

Comment:
maybe call it `particles_to_send`? how are the nested vectors organized, i.e., what do the indices mean?